src/pages/ExamsPage/components/AddAllowanceModal.jsx

```diff
@@ -0,0 +1,239 @@
+      pending: formatMessage(messages.addAllowanceButtonPendingLabel),
+      complete: formatMessage(messages.addAllowanceButtonCompleteLabel),
+      error: formatMessage(messages.addAllowanceButtonErrorLabel),
+    },
+    variant: 'primary',
+  };
+
+  const handleExamTypeChange = (examType) => {
+    if (examType === 'proctored') {
+      setDisplayExams(proctoredExams);
+    } else {
+      setDisplayExams(timedExams);
+    }
+    // reset exam selection state
+    setForm(prev => ({ ...prev, exams: {} }));
+  };
+
+  const handleChange = (event) => {
+    const el = event.target;
+    setForm(prev => ({ ...prev, [el.name]: el.value }));
+    if (el.name === 'exam-type') {
+      handleExamTypeChange(el.value);
+    }
+  };
+
+  const handleCheckboxChange = (event) => {
+    const el = event.target;
+    const id = el.getAttribute('data-key');
+    const timeLimitMins = el.value;
+    const selectedExams = form.exams;
+
+    if (el.checked) {
+      selectedExams[id] = timeLimitMins;
+    } else {
+      delete selectedExams[id];
+    }
+    setForm(prev => ({ ...prev, exams: selectedExams }));
+  };
+
+  const onSubmit = () => {
+    setLearnerFieldError(!form.users);
+    setExamFieldError(Object.keys(form.exams).length === 0);
+    setAdditionalTimeError(!form['additional-time-minutes'] && !form['additional-time-multiplier']);
+    if (form.users && form.exams && (form['additional-time-minutes'] || form['additional-time-multiplier'])) {
+      createAllowance(form);
+    }
+  };
+
+  return (
+    <ModalDialog
+      title="Add a new allowance"
+      isOpen={isOpen}
+      onClose={close}
+      size="md"
+      hasCloseButton
+      isFullscreenOnMobile
+    >
+      <ModalDialog.Header>
+        <ModalDialog.Title>
+          { formatMessage(messages.addAllowanceModalTitle) }
+        </ModalDialog.Title>
+      </ModalDialog.Header>
+      <ModalDialog.Body>
+        <Form>
+          <Form.Group isInvalid={learnerFieldError}>
+            <Form.Label>{ formatMessage(messages.addAllowanceLearnerField) }</Form.Label>
+            <Form.Control name="users" defaultValue="" onChange={handleChange} data-testid="users" />
+            { !learnerFieldError
+              ? (
+                <Form.Control.Feedback>
+                  { formatMessage(messages.addAllowanceLearnerFieldFeedback) }
+                </Form.Control.Feedback>
+              )
+              : (
+                <Form.Control.Feedback type="invalid">
+                  { formatMessage(messages.addAllowanceLearnerFieldErrorFeedback) }
+                </Form.Control.Feedback>
+              )}
+          </Form.Group>
+          <Form.Group controlId="form-exam-type">
+            <Form.Label>{ formatMessage(messages.addAllowanceExamTypeField) }</Form.Label>
+            <Form.Control
+              as="select"
+              onChange={handleChange}
+              name="exam-type"
+              defaultValue={form['exam-type']}
```

Comment:
I think since we are using `onChange()`, we need to use `value` here instead of `defaultValue`, otherwise we would have an hybrid controlled select, which is not ideal.